### PR TITLE
feat: Order conversation by priority

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://linear.app/chatwoot/issue/CW-1714/ui-sort-conversation-by-priority

https://github.com/chatwoot/chatwoot/assets/12408980/584e24e1-0038-4ca2-bdd1-eab3096ef716

### CODE REVIEW

File: app/javascript/dashboard/store/modules/conversations/getters.js [hunk 1]
```diff
@@ -13,6 +16,12 @@ const getters = {
     const comparator = {
       latest: (a, b) => b.last_activity_at - a.last_activity_at,
       sort_on_created_at: (a, b) => a.created_at - b.created_at,
+      sort_on_priority: (a, b) => {
+        return (
+          CONVERSATION_PRIORITY_ORDER[a.priority] -
+          CONVERSATION_PRIORITY_ORDER[b.priority]
+        );
+      },
```
Comment (scmmishra) [inline app/javascript/dashboard/store/modules/conversations/getters.js:24]: Why the need to add this here? Do we not get it from the server?

Comment (muhsin-k) [reply app/javascript/dashboard/store/modules/conversations/getters.js:24]: @scmmishra If we are adding any new conversation to the state via action cable events, we should also include the necessary sorting in the getter.

Comment (scmmishra) [reply app/javascript/dashboard/store/modules/conversations/getters.js:24]: Ahh, makes sense!

**Review by scmmishra — APPROVED** [review]
LGTM! just take a look at the failing build

**Pull request merged**